_posts/2023-08-26-gsoc23-final-report.md

```diff
@@ -0,0 +1,344 @@
+| [patch-hub: Fix bug and refactor 'Registered Mailing Lists' screen](https://github.com/kworkflow/kworkflow/pull/892) | 3 |
+| [src: ui: patch_hub: patch_hub_core: Fix 'Registered Mailing Lists' message box](https://github.com/kworkflow/kworkflow/pull/895) | 1 |
+
+### **Third Cycle: Consolidating Interaction with Lore API**
+
+After the two first cycles, we tackled what was considered from the onset of the program
+the critical point: the interactions with lore API, especially to fetch an arbitrary
+number of patchsets reliably, allowing the user to potentially navigate all of a mailing
+list history. It is important to note that, in case this problem couldn't be solved, the
+whole feature would be jeopardized as its functionality would be really limited.
+
+##### The Problem and the Solution
+
+I plan on making a more detailed post on the lore API, but in summary, lore provides a
+search engine powered by [Xapian](https://xapian.org/) that allows us to make queries
+to match specific messages in a given public mailing list archived.
+
+The implementation, at this point, used a hardcoded period of time (last 2 days) to
+query lore for patches and we needed a way to fetch adjacent chunks of patchsets that
+had a consistent order.
+
+After deeply studying the lore API, or, should I say, reverse-engineering it, I came
+up with an answer that both solved the problem and eliminated the need for managing
+timestamps to get consistent chunks of patchsets.
+
+##### Current Merged State of kw patch-hub
+
+All this blabber aside, below is a demo of using `kw patch-hub` to navigate through
+the *amd-gfx* list history. This demo is the current merged state of `kw patch-hub`.
+Notice that the feature paginates the patchsets and doesn't do redundant fetches when
+going back on pages.
+
+![kw patch-hub Current State]({{site.url}}/images/gifs/kw_patch_hub_current_state.gif)
+
+##### Contributions of Third Cycle
+
+This whole cycle is contained on this Pull Request with 9 commits that was active from
+the start of August until some days ago:
+
+[kw patch-hub: Add reliable fetch of latest patchsets from mailing list](https://github.com/kworkflow/kworkflow/pull/889)
+
+<br>
+
+## **Next Steps**
+
+---
+
+Considering the deliverables, I implemented number 1 and part of number 2. More
+specifically, `kw patch-hub` can be used as a somewhat reliable UI to the lore archives
+and provides some other functionalities like bookmarking patchsets, downloading
+applicable patchsets (to a default or custom directory), and managing the feature's
+settings through the feature itself.
+
+It's important to note that `kw patch-hub` has become an integral part of my Capstone
+Project, so I'll keep updating the feature until the end of this year, and probably
+further than that.
+
+Here is a list of practically all steps needed for `kw patch-hub` to be considered
+"complete", not necessarily in the order of importance:
```

Comment:
I'd prefer another perspective here. Think incrementally: `kw patch-hub` exists and can be used. As any active project, it will always require new features, improvements and maintenance, therefore you've mapped the next steps to enhance the tool and cover developers/user needs better. ;)

Reply:
For sure, I agree. This notion of "all steps needed" and "complete" isn't real. Thanks 😄